runtime/doc/map.txt

```diff
@@ -1431,6 +1431,128 @@ Possible values are (second column is the short name used in listing):
     -addr=other		  ?	other kind of range
 
 
+Incremental preview ~
+                                                  *:command-preview* {nvim-api}
+It's possible to allow commands to be incrementally previewed when
+'inccommand' is enabled by using the preview command attribute. The preview
+attribute may only be used through the API. (see |nvim_create_user_command()|)
+
+The value of the preview attribute must be either a boolean value, the name of
+a Vim script function or a Lua function callback. Here are some examples: >
+
+	-- setting preview to a boolean value
+	call nvim_create_user_command('foo', 'call Foo()',
+		\ { 'preview': v:true })
+
+	-- setting preview to a VimScript function
+	function! PreviewFunc()
+	    return v:true
+	endfunction
+
+	call nvim_create_user_command('bar', 'call Bar()',
+		\ { 'preview': 'PreviewFunc' })
+
+	-- setting preview to a Lua function callback
+	lua << EOF
+	local function PreviewFunc()
+	    return true
+	end
+
+	vim.api.nvim_create_user_command('baz', 'call Baz()',
+	                                 { preview = PreviewFunc })
+	EOF
+<
+
+If preview is a boolean value and its value is |TRUE|, preview is always
+triggered every time the command line is modified. If it's not a boolean value
+it calls the Vim script or Lua function and triggers command preview only if
+the function returns |TRUE|. The function may only take one argument, which is
+the contents of the command line.
+
+Every time a command needs to be previewed, Nvim executes the command. The
+command itself must do the job of making appropriate changes to the buffer
+and setting the appropriate highlights for previewing. It's possible to check
+if a command is being executed for previewing by checking if current mode is
+"cp" (see |mode()|).
+
+In order for command preview to work properly, the command must follow these
+steps when Nvim is in command live preview mode.
+    1. Make any necessary changes to the buffer for the preview
+    2. Add any necessary highlighting to the buffer
+    3. Trigger a redraw of the window (see |:redraw|)
```

Comment:
the docs have hyperlink-like functionality, which can often avoid redundant wording.

```suggestion
    3. Trigger a |:redraw| of the window
```